Null pointer thrown from addEditorOnboardingCommandId when keybinding without command exists

When a Keybinding can't find the command it belongs to, an NPE is thrown and the perspective can't be switched.
I am able to reproduce this with Eclipse 2023-06 and 2023-09

`Caused by: java.lang.NullPointerException: Cannot invoke "org.eclipse.e4.ui.model.application.commands.MCommand.getElementId()" because the return value of "org.eclipse.e4.ui.model.application.commands.MKeyBinding.getCommand()" is null
	at org.eclipse.ui.internal.e4.compatibility.ModeledPageLayout.lambda$1(ModeledPageLayout.java:647)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.tryAdvance(AbstractList.java:706)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:280)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.tryAdvance(AbstractList.java:706)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
	at org.eclipse.ui.internal.e4.compatibility.ModeledPageLayout.addEditorOnboardingCommandId(ModeledPageLayout.java:658)
	at org.eclipse.ui.internal.PerspectiveExtensionReader.processEditorOnboardingCommand(PerspectiveExtensionReader.java:421)
	at org.eclipse.ui.internal.PerspectiveExtensionReader.processExtension(PerspectiveExtensionReader.java:165)
	at org.eclipse.ui.internal.PerspectiveExtensionReader.readElement(PerspectiveExtensionReader.java:436)
	at org.eclipse.ui.internal.registry.RegistryReader.readElements(RegistryReader.java:123)
	at org.eclipse.ui.internal.registry.RegistryReader.readExtension(RegistryReader.java:133)
	at org.eclipse.ui.internal.registry.RegistryReader.readRegistry(RegistryReader.java:152)
	at org.eclipse.ui.internal.PerspectiveExtensionReader.extendLayout(PerspectiveExtensionReader.java:83)
	at org.eclipse.ui.internal.WorkbenchPage.createPerspective(WorkbenchPage.java:4116)
	at org.eclipse.ui.internal.WorkbenchPage.busySetPerspective(WorkbenchPage.java:4017)
	at org.eclipse.ui.internal.WorkbenchPage.lambda$12(WorkbenchPage.java:3977)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:67)
	at org.eclipse.ui.internal.WorkbenchPage.setPerspective(WorkbenchPage.java:3977)
	at org.eclipse.ui.internal.WorkbenchWindow.setup(WorkbenchWindow.java:899)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 81 more
 `

--- a/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java
+++ b/bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/e4/compatibility/ModeledPageLayout.java
@@ -20,11 +20,13 @@ import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 import java.util.function.Function;
 import java.util.function.Predicate;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.e4.ui.model.application.MApplication;
 import org.eclipse.e4.ui.model.application.commands.MBindingTable;
+import org.eclipse.e4.ui.model.application.commands.MCommand;
 import org.eclipse.e4.ui.model.application.commands.MKeyBinding;
 import org.eclipse.e4.ui.model.application.descriptor.basic.MPartDescriptor;
 import org.eclipse.e4.ui.model.application.ui.MElementContainer;
@@ -644,7 +646,9 @@ public class ModeledPageLayout implements IPageLayout {
 		if (numberOfOnboardingCommands >= 5)
 			return;
 
-		Predicate<MKeyBinding> commandWithEqualId = b -> b.getCommand().getElementId().equals(commandId);
+		Predicate<MKeyBinding> commandWithEqualId = b -> Optional.of(b).map(MKeyBinding::getCommand)
+				.map(MCommand::getElementId).filter(elementId -> elementId.equals(commandId)).isPresent();
+
 		Function<MKeyBinding, String> toCommandText = b -> {
 			try {
 				return b.getCommand().getCommandName() + EDITOR_ONBOARDING_COMMAND_SEPARATOR

--- a/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/api/ModeledPageLayoutTest.java
+++ b/tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/api/ModeledPageLayoutTest.java
@@ -124,6 +124,22 @@ public class ModeledPageLayoutTest {
 		assertThat(getNumberOfOnboardingCommands(perspective.getTags()), is(5));
 	}
 
+	@Test
+	public void addEditorOnboardingCommandWhenBrokenKeyBindingExsists() throws Exception {
+		MBindingTable bindingTable = MCommandsFactory.INSTANCE.createBindingTable();
+		MKeyBinding binding = createBinding("unknownCommand", "Unknown", "M1+1+5");
+		binding.setCommand(null);
+		bindingTable.getBindings().add(binding);
+		bindingTable.getBindings().add(createBinding("org.eclipse.ui.window.quickAccess", "Find Actions", "M1+3"));
+
+		application.getBindingTables().add(bindingTable);
+
+		pageLayout.addEditorOnboardingCommandId("org.eclipse.ui.window.quickAccess");
+
+		assertThat(perspective.getTags(), hasItem(ModeledPageLayout.EDITOR_ONBOARDING_COMMAND + "Find Actions$$$"
+				+ KeySequence.getInstance("M1+3").format()));
+	}
+
 	private int getNumberOfOnboardingCommands(List<String> commands) {
 		return commands.stream().filter(t -> t.startsWith(EDITOR_ONBOARDING_COMMAND)).mapToInt(i -> 1).sum();
 	}

